stable/sonarqube/templates/deployment.yaml

```diff
@@ -35,6 +35,35 @@ spec:
       securityContext:
 {{ toYaml .Values.securityContext | indent 8 }}
       initContainers:
+        - name: chmod-volume-mounts
+          command:
+            - "sh"
+            - "-c"
+            - 'mkdir -p $(printf "/opt/sonarqube/${1-%s\n}" temp logs data extensions/downloads extensions/plugins/tmp extensions/plugins) &&
+               chown 999:999 -R $(printf "/opt/sonarqube/${1-%s\n}" temp logs data extensions/downloads extensions/plugins/tmp extensions/plugins)'
```

Comment:
@halkeye. This is interesting piece of code. I've two questions here:

-  How did you figure out that the user-id and group-id have to both `999`? Is it arbitrary? The `Dockerfile`  creates `sonarqube : sonarqube` as `usename : group`, so somehow the numbers you chose for the ID, have to _same_ to be as the ID of the user and group?
-  Why do you call `mkdir -p` on the paths? Do they not exist already, as they're mounted?

Reply:
I'm not sure what your expecting out of an already merged. When I first created the PR, 999 was hard coded, at some point that changed, but the merge didn't address it.

It took months of rebasing/merging/etc to get this patch to land, and I didn't end up using sonarcube, so I won't be creating a new patch for any fixes. I thought i'd be helpful, but i've gotten nothing but flack for it.

The owners of the helm chart can feel free to revert it, but I won't be bothering with it.

Reply:
@halkeye : Well, my questions were entirely out of curiosity, hoping that I'll learn from how you figured out the user-id and group-id. My questions have nothing to do with the PR or its status as such. In fact, it was my first comment on this repo (even the organization). Just trying to learn something from this piece of code... so that I could write better code (and think) myself.

Reply:
Sorry I sniped then, I've been getting a lot of slack about this.

at the time of the first PR attempt, 999 was used everywhere. Also the directories did not exist when it started up. It was september of last year, I don't really remember much else about it, other than this was what I had to do to fix it.

Reply:
It's perfectly alright.

Yes, it seems `999` is very OS specific. It happens to work for some OS. Found this:

- https://github.com/SonarSource/docker-sonarqube/issues/240#issuecomment-463962708

and then another one, which allows users to customize it, using two environment variables (`PUID` and `PGID`):

- https://github.com/SonarSource/docker-sonarqube/pull/246

Thanks a lot. Learnt lots of stuff in the last couple of days. I'm writing my own charts for sonarqube, and during that I found this.